daemons/ipa-slapi-plugins/ipa-pwd-extop/common.c

```diff
@@ -548,22 +548,25 @@ int ipapwd_CheckPolicy(struct ipapwd_data *data)
     pol.min_pwd_length = IPAPWD_DEFAULT_MINLEN;
 
     switch(data->changetype) {
-        case IPA_CHANGETYPE_ADMIN:
-            /* The expiration date needs to be older than the current time
-             * otherwise the KDC may not immediately register the password
-             * as expired. The last password change needs to match the
-             * password expiration otherwise minlife issues will arise.
-             */
-            data->timeNow -= 1;
-            data->expireTime = data->timeNow;
-            break;
         case IPA_CHANGETYPE_NORMAL:
             /* Find the entry with the password policy */
             ret = ipapwd_getPolicy(data->dn, data->target, &pol);
             if (ret) {
                 LOG_TRACE("No password policy, use defaults");
             }
             break;
+	case IPA_CHANGETYPE_ADMIN:
+            /* The expiration date needs to be older than the current time
+             * otherwise the KDC may not immediately register the password
+             * as expired. The last password change needs to match the
+             * password expiration otherwise minlife issues will arise.
+             */
+            data->timeNow -= 1;
+            data->expireTime = data->timeNow;
+
+	    /* let set the entry password property according to its
+	     * entry password policy
+	     */
```

Comment:
Hello, thanks for reviewing. 
You are right the removal of the break is intentional in order to retrieve the user password policy (ipapwd_getPolicy) like this was done in case of password update by 'directory manager'.

I agree the fix should clearly state that fallthrough is intentional. Will change the fix.